query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/PostgresClientFactory.java

```diff
@@ -58,32 +81,56 @@ public PostgresClient getPostgresClient(String clusterName) {
 
   public static class PostgresClient {
 
-    private final Connection connection;
+    private final DataSource dataSource;
 
     private PostgresClient(RequestHandlerClientConfig clientConfig) throws SQLException {
       String user = clientConfig.getUser().orElseThrow(IllegalArgumentException::new);
       String password = clientConfig.getPassword().orElseThrow(IllegalArgumentException::new);
-      String url =
-          String.format(
-              "%s?user=%s&password=%s", clientConfig.getConnectionString(), user, password);
-      LOG.debug(
-          "Trying to create a Postgres client connected to postgres server using url: {}", url);
-      this.connection = DriverManager.getConnection(url);
+      String url = clientConfig.getConnectionString();
+      this.dataSource = createPooledDataSource(url, user, password);
     }
 
-    private PostgresClient(Connection connection) {
-      this.connection = connection;
+    private DataSource createPooledDataSource(String url, String user, String password) {
+      LOG.debug(
+          "Trying to create a Postgres client connected to postgres server using url: {}, user: {}, password: {}",
```

Comment:
Debug logs have more flexibility, but is there really value in logging the password here?

Reply:
Will remove the password

Reply:
Done